packages/git/src/browser/history/git-history-widget.ts

```diff
@@ -119,6 +109,21 @@ export class GitHistoryWidget extends GitBaseWidget<GitHistoryListNode> implemen
         }
     }
 
+    protected async addFileChangeNodesToCommit(commit: GitCommitNode) {
```

Comment:
The method should check whether `fileChangeNodes` have been computed already. It looks like it would rerender when collapse/expand multiple times even with duplicate entries? We could delete the `fileChanges` property when computed.